builtin/common/misc_helpers.lua

```diff
@@ -561,6 +561,21 @@ function table.insert_all(t, other)
 end
 
 
+function table.shuffle(t, from, to, rank)
+	from = from or 1
+	to = to or #t
+	local n = to - from + 1
+	rank = rank or math.random(0, math.faculty(n)-1)
```

Comment:
when shuffling a list of 40 things, `40!` would be larger than the largest possible float (3.4E+38)
this function won't work correct in that case

is there any reason you didn't just implement a "plain" shuffle?

Reply:
When putting items in a chest in random order during mapgen, the order should be the same if the chest is generated again at the same position with the same seed. I didn't consider the rank's size.